sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -134,10 +142,26 @@ private void startTracing(final @NonNull Activity activity) {
 
       // in case appStartTime isn't available, we don't create a span for it.
       if (firstActivityCreated || appStartTime == null) {
-        transaction = hub.startTransaction(activityName, UI_LOAD_OP, (Date) null, true);
+        transaction =
+            hub.startTransaction(
+                activityName,
+                UI_LOAD_OP,
+                (Date) null,
+                true,
+                (finishingTransaction) -> {
+                  activityFramesTracker.setMetrics(activity, finishingTransaction.getEventId());
+                });
```

Comment:
I'm probably missing something but couldn't we remove the `if` (and the `else` branch) and just pass `appStartTime` regardless if it's null or not? 

```suggestion
        transaction =
            hub.startTransaction(
                activityName,
                UI_LOAD_OP,
                appStartTime,
                true,
                (finishingTransaction) -> {
                  activityFramesTracker.setMetrics(activity, finishingTransaction.getEventId());
                });
```

Reply:
there are 2 conditions, `firstActivityCreated || appStartTime == null`, `appStartTime` might not be null and `firstActivityCreated` true so it'd give a wrong transaction start time.